CRM/Civicase/Helper/CaseCategory.php

```diff
@@ -8,8 +8,24 @@
  */
 class CRM_Civicase_Helper_CaseCategory {
 
+  const CASE_TYPE_CATEGORY_GROUP_NAME = 'case_type_categories';
   const CASE_TYPE_CATEGORY_NAME = 'Cases';
 
+  /**
+   * Returns the full list of case type categories.
+   *
+   * @return array
+   *   a list of case categories as returned by the option value API.
+   */
+  public static function getCaseCategories() {
```

Comment:
Thanks a lot @tunbola ! I have implemented `CaseType::buildOptions` and removed this helper function.

In all honesty, `CaseType::buildOptions` does not make any sense to me, the parameters are also weird. I'd recommend building a decorator that can be used instead of this. Something that makes more sense. But I'll leave that out for discussion, in the meantime I have implemented it the same as we are doing everywhere else.